test/specs/spec.js

```diff
@@ -271,24 +307,8 @@ describe('Swagger spec', function() {
                 if ({}.hasOwnProperty.call(mSpec, 'x-monitor') && !mSpec['x-monitor']) {
                     return;
                 }
-                validateExamples(pathStr, defParams, mSpec['x-amples']);
-            });
-        });
-    });
-
-    describe('routes', () => {
-
-        constructTests(spec.paths, defParams).forEach((testCase) => {
-            it(testCase.title, () => {
-                return preq(testCase.request)
-                .then((res) => {
-                    validateTestResponse(testCase, res);
-                }, (err) => {
-                    validateTestResponse(testCase, err);
-                });
+                validateExamples(pathStr, spec['x-default-params'] || {}, mSpec['x-amples']);
             });
         });
-
     });
 });
```

Comment:
Let's not just rename the test so that it's executed last - that's really bad to build upon a very unclear undocumented conincidence. See https://github.com/wikimedia/restbase/pull/1146 for a more appropriate solution (still imperfect though)

In general, I see a lot of changes here. What happened?

Reply:
TLDR: I applied the Restbase changes. The test failures being report incorrectly is still an issue.

Interdependence is easily addressed. We just need to require global.js from each test and everything works as expected.  

The multiple config issue was not addressed in the original server either so I didn't think we were losing any flexibility here. The original solution would just start the last service in the config. Looks like I am still appreciating the finer points here. Let's talk about it tomorrow.

The last issue had to do with asserts clearly failing yet the test was recorded as a success. I added the delay to leave the server up long enough for all the tests to finish. What I tried to point out was that if you shut the server down right away in after() some of the dynamic tests will clearly fail (expected 500 but got 504) but all 43 tests reported successes.

Reply:
> TLDR: I applied the Restbase changes. The test failures being reported incorrectly is still an issue.

See inline. 

 > The multiple config issue was not addressed in the original server either so I didn't think we were losing any flexibility here. The original solution would just start the last service in the config. 

Multiple configs !== multiple services. We can have `config.dev.yaml`, `config.thirdparty.yaml` and bla and we want to be able to write different test suites that use different configs. It indeed was not possible before, but our goal is to improve things, not maintain the status quo :)

> The last issue had to do with asserts clearly failing yet the test was recorded as a success. I added a delay to leave the server up long enough for all the tests to finish. What I tried to point out was that if you shut the server down right away in after() some of the dynamic tests will clearly fail (expected 500 but got 504 because the server is gone) but all 43 tests reported successes.

See inline.

Reply:
I suppose I didn't describe it well. While I agree we don't just want to maintain the status quo, that should the be part of the ticket :)